src/key_exchange/group/curve25519.rs

```diff
@@ -96,7 +96,7 @@ impl KeGroup for Curve25519 {
         bytes
             .try_into()
             .ok()
-            .and_then(|bytes| Scalar::from_canonical_bytes(bytes).into())
+            .map(Scalar::from_bits_clamped)
```

Comment:
For some reason `curve25519-dalek` does not like the keys supplied here. If I understood the `from_bits_clamped()` documentation correctly, it means that the supplied private keys are actually not valid and had to be clamped.

(IANAC and I have no clue what I'm talking about here)

Reply:
Clearly I don't understand what clamping actually means, but for some reason this breaks our current tests which we already generated test vectors for. Meaning clamping here does produce different keys and the keys we were producing until now are wrong?

Reply:
It does, I was surprised that `curve25519_dalek::Scalar` just is not safe to use then with Curve25519. In any case, I adjusted the code.

I still only got one more test working, the rest fails and I honestly have no clue.

About the test vectors, would it not be better if the test vectors include already pre-clamped private keys? Because I would otherwise say that deserializing private keys should not clamp, it should fail if the key is not valid.

Reply:
Yes, let me add a comment to the test vector PR for exactly this. Thanks!